lua/autorun/client/cl_mcsp.lua

```diff
@@ -0,0 +1,98 @@
+local MCSPmodel = "models/milaco/minecraft_pm/minecraft_pm.mdl"
+CreateClientConVar("mcsp_url", "", true, true)
+
+local function MCSPRenderNewMaterial(url, ply) --This is where the magic happens, baby
+	ply.MCSPSkinMaterial = ImgurMaterial(url, ply, ply:GetPos(), false, "VertexLitGeneric", {
```

Comment:
You dont' need to cache this. Just call ImgurMaterial right in PrePlayerDraw. It expects you to keep calling it and it will return different materials as the skin loads. It needs you to keep calling it so it knows when the material is no longer in use as well.

Reply:
The function also internally caches and reuses all the data so like, it's fast enough to call every frame with no issue at all.